portia-entrypoint

```diff
@@ -143,18 +145,26 @@ def is_zipunsafe(eggpath):
     return dist.has_metadata('not-zip-safe')
 
 
-def unpack_egg(eggpath):
-    newpath = os.path.join(DASH_UNPACKED_EGGS_PATH, os.path.basename(eggpath))
-    unpack_archive(eggpath, newpath)
-    os.remove(eggpath)
-    return newpath
+def unpack_egg(egg_path):
+    destination_parts = [os.path.basename(egg_path)]
+    is_addon_egg = os.path.dirname(egg_path) == DASH_ADDONS_EGGS_PATH
+    if is_addon_egg:
+        destination_parts.insert(0, 'addons')
+    destination = os.path.join(DASH_UNPACKED_EGGS_PATH, *destination_parts)
+    unpack_archive(egg_path, destination)
+    # we don't have permissions to drop files from addons folder
+    if not is_addon_egg:
+        os.remove(egg_path)
+    return destination
 
 
 def update_environment_for_portia():
     """Modify current environment for Portia."""
     env = os.environ.copy()
     env['TERM'] = 'xterm'
-    eggs_environment = load_dash_eggs(DASH_PACKED_EGGS_PATH)
+    # load addons_eggs in the end to give them precedence
+    eggs_environment = load_dash_eggs([DASH_PACKED_EGGS_PATH,
```

Comment:
We also have to load addons eggs built-in to the stack, the only question is whether we should allow to replace the addons eggs with custom eggs or not (it's defined by precedence of the paths here). Assuming that we're moving away from eggs completely, I think it's fine to keep it as-is, but let me know if you disagree